crates/nu-lsp/src/workspace.rs

```diff
@@ -46,6 +46,20 @@ fn find_nu_scripts_in_folder(folder_uri: &Uri) -> Result<nu_glob::Paths> {
     nu_glob::glob(&pattern, Uninterruptible).into_diagnostic()
 }
 
+/// HACK: when current file is imported (use keyword) by others in the workspace,
+/// it will get parsed a second time via `parse_module_block`, so that its definitions'
+/// ids are renewed, making it harder to track the references.
+/// The fake path stops that with `ParseError::ModuleNotFound`,
+/// which seems benign for tasks like references/renaming.
+///
+/// FIXME: cross-file shadowing can still cause false-positive/false-negative cases
```

Comment:
This note explains the bug mentioned [here](https://github.com/nushell/nushell/pull/15470#issuecomment-2771804887).

I've tried other methods like creating a module from existing block, but they ended up making it worse (either crash or totally wrong spans).
This hack, although seems dangerous, actually behaves closest to ideal according to my tests.

I'd like to leave it in this PR as a stopgap while searching for better solutions.

Reply:
I'm not opposed to it but hope you find something better too.